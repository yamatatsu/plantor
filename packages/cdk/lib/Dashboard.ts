import { App, Stack, StackProps, aws_cloudwatch, Duration } from "aws-cdk-lib";

type Props = StackProps & {};

export class Dashboard extends Stack {
  constructor(parent: App, id: string, props: Props) {
    super(parent, id, props);

    new aws_cloudwatch.Dashboard(this, "Dashboard", {
      dashboardName: "plantor",
      start: "-P1W",
      widgets: [
        [
          new aws_cloudwatch.GraphWidget({
            title: "Moisture",
            width: 24,
            left: [
              new aws_cloudwatch.Metric({
                metricName: "001",
                namespace: "CUSTOM-IoT/Moisture",
                period: Duration.hours(1),
                statistic: "Average",
              }),
            ],
          }),
        ],
      ],
    });
  }
}
