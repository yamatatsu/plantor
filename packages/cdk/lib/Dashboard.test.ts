import { SynthUtils } from "@aws-cdk/assert";
import { App } from "aws-cdk-lib";
import { Dashboard } from "./Dashboard";

test("snapshot test", () => {
  const app = new App();

  const target = new Dashboard(app, "Target", {});

  expect(SynthUtils.toCloudFormation(target)).toMatchSnapshot();
});
