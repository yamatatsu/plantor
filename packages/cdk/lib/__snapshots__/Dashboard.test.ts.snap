// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "Resources": Object {
    "Dashboard9E4231ED": Object {
      "Properties": Object {
        "DashboardBody": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"start\\":\\"-P1M\\",\\"widgets\\":[{\\"type\\":\\"metric\\",\\"width\\":6,\\"height\\":3,\\"x\\":0,\\"y\\":0,\\"properties\\":{\\"view\\":\\"singleValue\\",\\"title\\":\\"moisture001\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[{\\"label\\":\\"moisture001\\",\\"expression\\":\\"((m1 - 640) / (380 - 640)) * 100\\",\\"period\\":21600}],[\\"CUSTOM-IoT/Moisture\\",\\"moisture001\\",{\\"period\\":21600,\\"visible\\":false,\\"id\\":\\"m1\\"}]]}},{\\"type\\":\\"metric\\",\\"width\\":24,\\"height\\":6,\\"x\\":0,\\"y\\":3,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Moisture\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"metrics\\":[[{\\"label\\":\\"moisture001\\",\\"expression\\":\\"((m1 - 640) / (380 - 640)) * 100\\",\\"period\\":21600}],[\\"CUSTOM-IoT/Moisture\\",\\"moisture001\\",{\\"period\\":21600,\\"visible\\":false,\\"id\\":\\"m1\\"}]],\\"yAxis\\":{}}}]}",
            ],
          ],
        },
        "DashboardName": "CloudWatch-Default",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
  },
}
`;
